src/Effects/Chorus.cpp

```diff
@@ -131,41 +176,69 @@ void Chorus::out(const Stereo<float *> &input)
         inR = tmpc.r * (1.0f - lrcross) + tmpc.l * lrcross;
 
         //Left channel
-
-        //compute the delay in samples using linear interpolation between the lfo delays
-        float mdel =
-            (dl1 * (buffersize - i) + dl2 * i) / buffersize_f;
+        // reset output accumulator
+        output = 0.0f;
+        // increase delay line writing position and handle turnaround
         if(++dlk >= maxdelay)
             dlk = 0;
-        float tmp = dlk - mdel + maxdelay * 2.0f; //where should I get the sample from
-
-        dlhi  = (int) tmp;
-        dlhi %= maxdelay;
-
-        float dlhi2 = (dlhi - 1 + maxdelay) % maxdelay;
-        float dllo  = 1.0f + floorf(tmp) - tmp;
-        efxoutl[i] = cinterpolate(delaySample.l, maxdelay, dlhi2) * dllo
-                     + cinterpolate(delaySample.l, maxdelay,
-                                    dlhi) * (1.0f - dllo);
-        delaySample.l[dlk] = inL + efxoutl[i] * fb;
+        // linear interpolate from old to new value over length og the buffer
+        float dl = (dlHist * (buffersize - i) + dlNew * i) / buffersize_f;
+        // get sample with that delay form delay line and add to output accumulator
+        output += getSample(delaySample.l, dl, dlk);
+        switch (Pflangemode) {
+            case DUAL:
+            // calculate and apply delay for second ensemble member
+            dl = (dlHist2 * (buffersize - i) + dlNew2 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+                break;
+            case TRIPLE:
+            // calculate and apply delay for second ensemble member
+            dl = (dlHist2 * (buffersize - i) + dlNew2 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+            // same for third ensemble member
+            dl = (dlHist3 * (buffersize - i) + dlNew3 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+            // reduce amplitude to match single phase modes
+            output *= 0.85f;
+                break;
+            default:
+                // nothing to do for standard chorus
+                break;
+        }
+        // store current input + feedback to delay line at writing position
+        delaySample.l[dlk] = inL + output * fbComp;
```

Comment:
Sometimes, I think we could just add `operator=`, `operator*` etc. to class `delaySample` (thought I guess it would be easier to use `std::complex` or similar?) and then just write the code once, resulting in `in`, `output` and `delaySample` holding left and right samples, resulting in code like
``` c++
delaySample[dlk] = in + output * fbComp;
```
Though I am fine with the current code, too.

Reply:
that's too difficult for me. but i was thinking about refactoring it to a function that is called twice

Reply:
Wouldn't `std::complex` do the job easily?

@fundamental Do you have an opinion here?

Reply:
I vote for merging this functional change and use a second PR for refactoring stuff.

Reply:
I'm content with refactoring getting pushed into a later PR

Reply:
Resolving this.